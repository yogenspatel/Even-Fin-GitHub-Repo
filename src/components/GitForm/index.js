/* eslint-disable jsx-a11y/label-has-for */
/* eslint-disable jsx-a11y/label-has-associated-control */
// FormCode.js

import React from 'react';
import { Field, reduxForm } from 'redux-form';
import PropTypes from 'prop-types';

const validate = values => {
  const errors = {};
  if (!values.searchText) {
    errors.searchText = 'Required';
  } else if (values.searchText.length < 2) {
    errors.searchText = 'Minimum be 2 characters or more';
  }
  return errors;
};

const renderField = ({ input, label, type, meta: { touched, error, warning } }) => (
  <div>
    <input {...input} placeholder={label} type={type} className="form-control" />
    {touched &&
      ((error && <span className="text-danger">{error}</span>) ||
        (warning && <span>{warning}</span>))}
  </div>
);

// eslint-disable-next-line import/no-mutable-exports
let GitForm = props => {
  const { handleSubmit, pristine, submitting } = props;

  return (
    <form onSubmit={handleSubmit} className="container">
      <div className="row">
        <div className="form-group col">
          <Field name="searchText" component={renderField} label="Search Text" />
        </div>
        <div className="form-group col">
          <Field name="stars" component={renderField} label="stars" />
        </div>
      </div>
      <div className="row">
        <div className="form-group col">
          <label>Licence</label>:
          <Field name="licence" component="select" className="custom-select custom-select-lg mb-3">
            <option value="mit">MIT</option>
            <option value="isc">ISC</option>
            <option value="apache">Apache</option>
            <option value="gpl">GPL</option>
          </Field>
        </div>
        <div className="col custom-control custom-checkbox">
          <Field name="hasForked" component="input" type="checkbox" /> <label>Include Forked</label>
        </div>
      </div>
      <div className="col text-center form-group">
        <button type="submit" disabled={pristine || submitting} className="btn btn-primary">
          SEARCH
        </button>
      </div>
    </form>
  );
};

GitForm.propTypes = {
  handleSubmit: PropTypes.func,
  pristine: PropTypes.bool,
  submitting: PropTypes.bool,
};

GitForm.defaultProps = {
  handleSubmit: () => {},
  pristine: false,
  submitting: false,
};

GitForm = reduxForm({
  form: 'gitform',
  validate
})(GitForm);

export default GitForm;
