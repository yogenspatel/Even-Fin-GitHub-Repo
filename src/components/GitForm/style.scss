.custom-checkbox {
    margin-top: 50px;
}
.custom-checkbox input {
    -ms-transform: scale(2); /* IE */
    -moz-transform: scale(2); /* FF */
    -webkit-transform: scale(2); /* Safari and Chrome */
    -o-transform: scale(2); /* Opera */
    padding: 10px;
}

.custom-checkbox label {
    margin-left: 10px;
}
