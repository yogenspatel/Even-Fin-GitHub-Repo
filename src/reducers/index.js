import { createStore, combineReducers } from "redux";
import { reducer as reduxFormReducer } from "redux-form";
import searchReducer from "./search";

const rootReducer = combineReducers({
  form: reduxFormReducer, // mounted under "form"
  searchResults: searchReducer
});

function configureStore(state = {}) {
  return createStore(rootReducer, state);
}

export { configureStore };
export default rootReducer;
