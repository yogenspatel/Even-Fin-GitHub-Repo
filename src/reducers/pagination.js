/* eslint-disable no-case-declarations */
import { PAGINATED_DATA } from "../actions";

export default (state = null, action) => {
  switch (action.type) {
    case PAGINATED_DATA:
      const { payload, noOfItems, pageSize, searchData } = action;
      return { ...state, paginated_data: payload, noOfItems, pageSize, searchData };
    default:
      return state;
  }
};
