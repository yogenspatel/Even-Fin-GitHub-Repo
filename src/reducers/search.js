import { PERFORM_SEARCH } from '../actions';

export default (state = null, action) => {
  switch (action.type) {
    case PERFORM_SEARCH:
      return action.payload.data.items;
    default:
      break;
  }
  return state;
};
