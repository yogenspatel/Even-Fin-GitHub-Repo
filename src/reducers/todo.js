import { GET_TODO_DATA, SET_LOADING_STATUS, SET_DATA_ERROR, SORT_DATA } from "../actions";

export default (state = null, action) => {
  switch (action.type) {
    case SET_LOADING_STATUS:
      return { ...state, status: action.status };
    case SET_DATA_ERROR:
      return { ...state, status: action.error };
    case GET_TODO_DATA:
      return action.payload ? action.payload : state;
    case SORT_DATA:
      console.log('state: ', action.sort_data);
      return action.sort_data;
    default:
      return state;
  }
};
