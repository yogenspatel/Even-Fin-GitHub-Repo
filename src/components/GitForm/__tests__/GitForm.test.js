/* eslint-disable no-undef */

import React from "react";
import GitForm from "../../GitForm";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";
import { Provider } from "react-redux";

import { validate } from "../../GitForm";
import { configureStore } from "../../../reducers";

it("Git Form renders without crashing", () => {
  const store = configureStore();
  const tree = renderer
    .create(
      <Provider store={store}>
        <MemoryRouter>
          <GitForm handleSubmit={() => {}} />
        </MemoryRouter>
      </Provider>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it("Validates search Text", () => {
  const validFn = validate({ searchText: "test Search Text" });
  expect(validFn).toEqual({});
});

it("Validates search Text with less than 2 chars", () => {
  const validFn = validate({ searchText: "t" });
  expect(validFn).toEqual({ searchText: "Minimum be 2 characters or more" });
});

it("Validates search Text when no search term entered", () => {
  const validFn = validate({ searchText: "" });
  expect(validFn).toEqual({ searchText: "Required" });
});
