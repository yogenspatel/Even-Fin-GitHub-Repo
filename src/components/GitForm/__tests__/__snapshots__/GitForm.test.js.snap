// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Git Form renders without crashing 1`] = `
<form
  className="container"
  onSubmit={[Function]}
>
  <div
    className="row"
  >
    <div
      className="form-group col"
    >
      <div>
        <input
          className="form-control"
          name="searchText"
          onBlur={[Function]}
          onChange={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          placeholder="Search Text"
          value=""
        />
      </div>
    </div>
    <div
      className="form-group col"
    >
      <div>
        <input
          className="form-control"
          name="stars"
          onBlur={[Function]}
          onChange={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          placeholder="stars"
          value=""
        />
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="form-group col"
    >
      <label>
        License
      </label>
      :
      <select
        className="custom-select custom-select-lg mb-3"
        name="license"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        value=""
      >
        <option
          value="mit"
        >
          MIT
        </option>
        <option
          value="isc"
        >
          ISC
        </option>
        <option
          value="apache"
        >
          Apache
        </option>
        <option
          value="gpl"
        >
          GPL
        </option>
      </select>
    </div>
    <div
      className="col custom-control custom-checkbox"
    >
      <input
        checked={false}
        name="hasForked"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        type="checkbox"
        value=""
      />
       
      <label>
        Include Forked
      </label>
    </div>
  </div>
  <div
    className="col text-center form-group"
  >
    <button
      className="btn btn-primary"
      disabled={true}
      type="submit"
    >
      SEARCH
    </button>
  </div>
</form>
`;
